database/cassandra.js

```diff
@@ -47,9 +48,82 @@ client.connect()
 
 .then(() => {
   console.log('success');
-  return client.shutdown();
+  // ---------------------------------- Note: ----------------------------------
+  // might have to shutdown at some point to free up connections
+  // return client.shutdown();
 })
 .catch((err) => {
   console.error('there was an error:', err);
   return client.shutdown().then(() => { throw err; });
 })
+
+var save = (review, hotelId) => {
+
+  return new Promise((resolve, reject) => {
+    // save expects a review in the format shown at the bottom
+    const query = `INSERT INTO reviews.hotels1 \
+    (id,hotelid,responderorg,responderpicture,responderclose,responderdate,respondername,\
+      responderposition,respondertext,memberid,memberimg,memberusername,memberlocation,\
+      membercontributions,memberhelpful,reviewdate,reviewtitle,reviewtext,reviewtriptype,\
+      reviewpictures,reviewratings) VALUES \
+      (${review.id},${review.hotelId},'${review.responderOrg}','${review.responderPicture}',\
+      '${review.responderClose}','${review.responderDate}','${review.responderName}',\
+      '${review.responderPosition}','${review.responderText}',${review.memberId},\
+      '${review.memberImg}','${review.memberUserName}','${review.memberLocation}',\
+      ${review.memberContributions},${review.memberHelpful},'${review.reviewDate}',\
+      '${review.reviewTitle}','${review.reviewText}','${review.reviewTripType}',\
+      '${review.reviewPictures}',${review.reviewRatings}
+      )`;
```

Comment:
You could automate it like this:
```
const columns = [all_column_names];
let query = 'INSERT INTO reviews.hotels1 (';
let values = 'VALUES (';
for(let column of columns) {
  query.concat(column + ',');
  values.concat(review[column] + ',');
}
query.pop();
values.pop();
query += ') ' + values + ')';
```

That way you'd only have to change the array to update the full query. Probably not worth re-working it at this point, just thought I'd mention it.

Reply:
I like that! I might incorporate something like it